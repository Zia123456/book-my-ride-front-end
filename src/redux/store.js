import { configureStore } from '@reduxjs/toolkit';
import authReducer from './auth/authSlice';
import dataReducer from './reservation/reservationsListSlice';
import carReducer from './car/carSlice';
import addReservationReducer from './reservation/addReservationSlice';

const store = configureStore({
  reducer: {
    data: dataReducer,
    auth: authReducer,
    cars: carReducer,
    addReservation: addReservationReducer,
  },
});

export default store;
