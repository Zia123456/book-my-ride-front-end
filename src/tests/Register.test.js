import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/store';
import RegisterScreen from '../components/auth/Register';

it('Register component should render correctly', () => {
  const result = render(
    <React.StrictMode>
      <Router>
        <Provider store={store}>
          <RegisterScreen />
        </Provider>
      </Router>
    </React.StrictMode>,
  );
  expect(result).toMatchSnapshot();
});
