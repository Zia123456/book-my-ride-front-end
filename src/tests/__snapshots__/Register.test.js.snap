// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register component should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="registerPage"
      >
        <form
          class="form"
        >
          <div
            class="top"
          >
            <header>
              Sign Up
            </header>
            <div
              class="input-field reg-field"
            >
              <input
                id="name"
                name="name"
                placeholder="Name"
                required=""
                type="text"
                value=""
              />
            </div>
            <div
              class="input-field reg-field"
            >
              <input
                id="email"
                name="email"
                placeholder="Email"
                required=""
                type="email"
                value=""
              />
            </div>
            <div
              class="input-field reg-field"
            >
              <input
                id="password"
                name="password"
                placeholder="Password"
                required=""
                type="password"
                value=""
              />
            </div>
            <div
              class="input-field reg-field"
            >
              <input
                id="confirmPassword"
                name="confirmPassword"
                placeholder="confirmPassword"
                required=""
                type="password"
                value=""
              />
            </div>
            <button
              class="submit"
              type="submit"
            >
              Sign Up
            </button>
            <div
              class="two-col"
            >
              <div
                class="one"
              >
                <p>
                  Already have an account? 
                </p>
                <p>
                  <a
                    class="whiteLink"
                    href="/login"
                  >
                    Login Here
                  </a>
                </p>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="registerPage"
    >
      <form
        class="form"
      >
        <div
          class="top"
        >
          <header>
            Sign Up
          </header>
          <div
            class="input-field reg-field"
          >
            <input
              id="name"
              name="name"
              placeholder="Name"
              required=""
              type="text"
              value=""
            />
          </div>
          <div
            class="input-field reg-field"
          >
            <input
              id="email"
              name="email"
              placeholder="Email"
              required=""
              type="email"
              value=""
            />
          </div>
          <div
            class="input-field reg-field"
          >
            <input
              id="password"
              name="password"
              placeholder="Password"
              required=""
              type="password"
              value=""
            />
          </div>
          <div
            class="input-field reg-field"
          >
            <input
              id="confirmPassword"
              name="confirmPassword"
              placeholder="confirmPassword"
              required=""
              type="password"
              value=""
            />
          </div>
          <button
            class="submit"
            type="submit"
          >
            Sign Up
          </button>
          <div
            class="two-col"
          >
            <div
              class="one"
            >
              <p>
                Already have an account? 
              </p>
              <p>
                <a
                  class="whiteLink"
                  href="/login"
                >
                  Login Here
                </a>
              </p>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
